ballerina/listener_utils.bal

```diff
@@ -51,10 +51,10 @@ isolated function handlePostRequests(Engine engine, Context context, http:Reques
 
 isolated function getResponseFromJsonPayload(Engine engine, Context context, http:Request request,
                                              map<Upload|Upload[]> fileInfo = {}) returns http:Response {
-    var payload = request.getJsonPayload();
+    json|http:ClientError payload = request.getJsonPayload();
     if payload is json {
```

Comment:
Shall we change this logic a bit?

Check for the error first, return it, and then do other processing outside? We don't need to add the logic inside a block then.